PSSwagger/Test-FilteredResult.ps1

```diff
@@ -0,0 +1,72 @@
+Microsoft.PowerShell.Core\Set-StrictMode -Version Latest
+
+<#
+.DESCRIPTION
+   Determines if a result matches the given filter.
+
+.PARAMETER  Result
+    Result to filter
+
+.PARAMETER  Filter
+    Filter to apply
+#>
+function Test-FilteredResult {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [object]
+        $Result,
+
+        [Parameter(Mandatory=$true)]
+        [PSCustomObject]
+        $Filter
+    )
+
+    $ErrorActionPreference = 'Stop'
+    if ($Filter.Type -eq 'wildcard') {
+        Test-WildcardFilterOnResult -Filter $Filter -Result $Result
+    } elseif ($Filter.Type -eq 'logicalOperation') {
+        Test-LogicalFilterOnResult -Filter $Filter -Result $Result
+    }
+}
+
+function Test-WildcardFilterOnResult {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [object]
+        $Result,
+
+        [Parameter(Mandatory=$true)]
+        [PSCustomObject]
+        $Filter
+    )
+
+    $regex = $Filter.Value.Replace($Filter.Character, ".*")
+    ($Result.($Filter.Property)) -match $regex
+}
```

Comment:
Please consider using System.Management.Automation.WildcardPattern.

For example:
$inputNameWithOptionalWildcard = "Module*"
$resultName = "ModuleName"

$wildcardOptions = [System.Management.Automation.WildcardOptions]::CultureInvariant -bor [System.Management.Automation.WildcardOptions]::IgnoreCase
$wildcardPattern = New-Object System.Management.Automation.WildcardPattern $inputNameWithOptionalWildcard,$wildcardOptions
$wildcardPattern.IsMatch($resultName)

Reply:
Agreed to change default to '%' but option to change needs to stay (for cases where '%' is valid in the parameter string).

Reply:
70eadcb